app/controllers/api/v1/users/registrations_controller.rb

```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+
+module Api
+  module V1
+    module Users
+      class RegistrationsController < Devise::RegistrationsController
+        respond_to :json
+
+        before_action :ensure_valid_client
+
+        def create
+          user = User.from_email(user_params[:email], user_params[:password])
+          user.errors.empty? ? render_success(user) : render_error(user)
+        end
+
+        private
+
+        def user_params
+          params.require(:user).permit(:email, :password)
+        end
+
+        def render_success(user)
+          data = UserSerializer.new(user).serializable_hash[:data]
+          message = I18n.t('activemodel.notices.models.user.create')
```

Comment:
Please explicit the name to clarify the purpose. 😅 

```suggestion
          user_data = UserSerializer.new(user).serializable_hash[:data]
          meta_message = I18n.t('activemodel.notices.models.user.create')
```